db/migrate/20180130160640_add_unique_constraint_to_entries.rb

```diff
@@ -1,5 +1,5 @@
 class AddUniqueConstraintToEntries < ActiveRecord::Migration[5.1]
   def change
-    add_index(:entries, %i[hash_value entry_type timestamp spina_register_id], unique: true, name: 'unique_entry_index')
+    add_index(:entries, %i[hash_value entry_type entry_number spina_register_id], unique: true, name: 'unique_entry_index')
```

Comment:
I have a slight concern that this still leaves the possibility open to create a duplicate entry but with a different entry number.

Reply:
I'm not sure I understand what you mean here by a duplicate entry - could you elaborate please?

Reply:
where the other fields are the same but the entry number has been mis-calculated. As we have seen examples of this.